### PR TITLE
Support replace block and version with upper case

### PR DESCRIPTION
**Pull request by AlexeiVainshtein**

### CODE REVIEW

File: executers/dependenciesutils_test.go [hunk 1]
```diff
@@ -157,6 +159,112 @@ func TestMergeReplaceDependenciesWithGraphDependencies(t *testing.T) {
 	}
 }
 
+func TestParseModForReplaceDependencies(t *testing.T) {
```
Comment (eyalbe4) [inline executers/dependenciesutils_test.go]: The test is pretty hard to read as it is now.
Maybe we can move the mod content into a resource file?

File: executers/populateandpublish.go [hunk 2]
```diff
@@ -267,6 +267,9 @@ func (pwd *PackageWithDeps) getModPathInTemp(tempDir string) string {
 	moduleId := pwd.Dependency.GetId()
 	moduleInfo := strings.Split(moduleId, ":")
 	moduleInfo[0] = replaceExclamationMarkWithUpperCase(moduleInfo[0])
```
Comment (eyalbe4) [inline executers/populateandpublish.go]: I think that the name ```replaceExclamationMarkWithUpperCase``` is a bit too long...
I suggest we have two functions - one named ```goModEncode``` and one named ```goModDecode```.
Let's have the two functions next to each other in the same utility file.
We can also add a test that does encode and decode, and verifies that the outcome has not changed,

File: cmd/utils.go [hunk 3]
```diff
@@ -49,6 +49,14 @@ func prepareCmdOutputPattern() error {
 			}
 		}
 	}
+
+	if shouldInitRegExp && notFoundZipRegExp == nil {
+		log.Debug("Initializing not found zip file")
+		notFoundZipRegExp, err = initRegExp(`unknown import path ["]([^\/\r\n]+\/[^\r\n\s:]*)["].*(404( Not Found)?[\s]?)$`, Error)
+		if err != nil {
+			return err
+		}
```
[the remark this answers is not in the document]
Comment (AlexeiVainshtein) [reply cmd/utils.go]: Done

File: cmd/utils.go [hunk 4]
```diff
@@ -72,7 +80,10 @@ func MaskCredentials(pattern *gofrogio.CmdOutputPattern) (string, error) {
 }
 
 func Error(pattern *gofrogio.CmdOutputPattern) (string, error) {
-	fmt.Fprintf(os.Stderr, pattern.Line)
+	_, err := fmt.Fprint(os.Stderr, pattern.Line)
```
[the remark this answers is not in the document]
Comment (AlexeiVainshtein) [reply cmd/utils.go]: I've added the CheckError

File: executers/populateandpublish.go (hunk 2 above)
Comment (AlexeiVainshtein) [reply executers/populateandpublish.go]: Done I will add a test

File: executers/dependenciesutils.go [hunk 5]
```diff
@@ -263,9 +263,9 @@ func GetDependencies(cachePath string, moduleSlice map[string]bool) ([]Package,
 // Returns the actual path to the dependency.
 // If in the path there are capital letters, the Go convention is to use "!" before the letter.
 // The letter itself in lowercase.
-func getDependencyName(name string) string {
+func getDependencyToLowerCase(upperCaseValue string) string {
```
[the remark this answers is not in the document]
Comment (AlexeiVainshtein) [reply executers/dependenciesutils.go]: I've changed to the goModEncode as suggested below.

File: cmd/utils.go [hunk 6]
```diff
@@ -15,7 +15,7 @@ import (
 )
 
 // Compiles all the regex once
-func prepareCmdOutputPattern() error {
+func prepareCmdOutputPattern(shouldInitRegExp bool) error {
```
[the remark this answers is not in the document]
Comment (AlexeiVainshtein) [reply cmd/utils.go]: Yes we can. I've changed this.

File: executers/dependenciesutils_test.go (hunk 1 above)
Comment (AlexeiVainshtein) [reply executers/dependenciesutils_test.go]: Done

File: executers/populateandpublish.go (hunk 2 above)
Comment (AlexeiVainshtein) [reply executers/populateandpublish.go]: Done added a test

File: executers/dependenciesutils.go [hunk 7]
```diff
@@ -154,17 +154,35 @@ func createDependencyInTemp(zipPath string) (tempDir string, err error) {
...
+		if unicode.IsUpper(letter) {
+			path += "!" + strings.ToLower(string(letter))
+		} else {
+			path += string(letter)
+		}
+	}
+	return path
+}
+
+// Returns the value with upper case.
```
Comment (eyalbe4) [inline executers/dependenciesutils.go]: The above comment line is incorrect grammatically.

Comment (AlexeiVainshtein) [reply executers/dependenciesutils.go]: Changed it.

**Pull request merged**